pytorch_lightning/strategies/ddp.py

```diff
@@ -422,8 +426,9 @@ def teardown(self) -> None:
         if isinstance(self.model, DistributedDataParallel):
             self.model = self.lightning_module
 
-        if self.sync_batchnorm:
-            self.model = _revert_sync_batchnorm(self.model)
+        if self.lightning_module.trainer:
+            if self.lightning_module.trainer.state.fn == TrainerFn.FITTING and self.sync_batchnorm:
+                self.model = _revert_sync_batchnorm(self.model)
```

Comment:
nit: these can go together

```suggestion
        if self.lightning_module.trainer and self.lightning_module.trainer.state.fn == TrainerFn.FITTING and self.sync_batchnorm:
            self.model = _revert_sync_batchnorm(self.model)
```